client/extensions/wp-job-manager/state/data-layer/settings/index.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { displaySettings, enableSettings } from '../../settings/actions';
+import { WP_JOB_MANAGER_FETCH_SETTINGS } from 'wp-job-manager/state/action-types';
+
+export const fetch = ( { dispatch }, action ) => {
+	const { siteId } = action;
+
+	dispatch( http( {
+		method: 'GET',
+		path: `/jetpack-blogs/${ siteId }/rest-api/`,
+		query: {
+			path: '/wpjm/v1/settings',
+		},
+	}, action ) );
+};
+
+export const displayAndEnable = ( { dispatch }, { siteId }, next, { data } ) => {
+	dispatch( displaySettings( siteId, data ) );
+	dispatch( enableSettings( siteId ) );
+};
+
+export const enable = ( { dispatch }, { siteId } ) => dispatch( enableSettings( siteId ) );
+
+const dispatchSettingsRequest = dispatchRequest( fetch, displayAndEnable, enable );
```

Comment:
Agree with all your points. Fixed in https://github.com/Automattic/wp-calypso/pull/15299/commits/2ace8f9af117f9fec9f2ac63acdc4c2dd0c2d427.